addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/handler/IPBridgeThingHandler.java

```diff
@@ -0,0 +1,117 @@
+
+    public IPBridgeThingHandler(Thing thing, ItemChannelLinkRegistry itemChannelLinkRegistry) {
+        super(thing, itemChannelLinkRegistry);
+    }
+
+    public IPBridgeThingHandler() {
+        // Default constructor required in order to have OSGi Declarative Services working
+        super(null, null);
+    }
+
+    @Override
+    public void initialize() {
+
+        ip = (String) getConfig().get(IP_ADDRESS);
+        localsource = (String) getConfig().get(LOCAL_SOURCE_ADDRESS);
+
+        String connectionTypeString = (String) getConfig().get(IP_CONNECTION_TYPE);
+        if (StringUtils.isNotBlank(connectionTypeString)) {
+            if ("TUNNEL".equals(connectionTypeString)) {
+                ipConnectionType = KNXNetworkLinkIP.TUNNELING;
+            } else if ("ROUTER".equals(connectionTypeString)) {
+                ipConnectionType = KNXNetworkLinkIP.ROUTING;
+                if (StringUtils.isBlank(ip)) {
+                    ip = KNXBindingConstants.DEFAULT_MULTICAST_IP;
+                }
+                if (StringUtils.isBlank(localsource)) {
+                    localsource = KNXBindingConstants.DEFAULT_LOCAL_SOURCE_ADDRESS;
+                }
+            } else {
+                logger.warn("unknown IP connection type '" + connectionTypeString
+                        + "'! Known types are either 'TUNNEL' or 'ROUTER'");
+            }
+        } else {
+            ipConnectionType = KNXNetworkLinkIP.TUNNELING;
+        }
+
+        super.initialize();
+    }
+
+    @Override
+    public void establishConnection() throws KNXException {
+
+        try {
+            InetSocketAddress localEndPoint = null;
+            if (StringUtils.isNotBlank((String) getConfig().get(LOCAL_IP))) {
+                localEndPoint = new InetSocketAddress((String) getConfig().get(LOCAL_IP), 0);
+            } else {
+                try {
+                    InetAddress localHost = InetAddress.getLocalHost();
+                    localEndPoint = new InetSocketAddress(localHost, 0);
+                } catch (UnknownHostException uhe) {
+                    logger.warn(
+                            "Couldn't find an IP address for this host. Please check the .hosts configuration or use the 'localIp' parameter to configure a valid IP address.");
+                }
+            }
+
+            if (logger.isInfoEnabled()) {
+                if (link instanceof KNXNetworkLinkIP) {
+                    String ipConnectionTypeString = ipConnectionType == KNXNetworkLinkIP.ROUTING ? "ROUTER" : "TUNNEL";
+                    logger.info("Establishing connection to KNX bus on {} in mode {}.",
+                            ip + ":" + ((BigDecimal) getConfig().get(PORT_NUMBER)).intValue(), ipConnectionTypeString);
+                }
+            }
+
+            link = new KNXNetworkLinkIP(ipConnectionType, localEndPoint,
+                    new InetSocketAddress(ip, ((BigDecimal) getConfig().get(PORT_NUMBER)).intValue()), false,
```

Comment:
Suggestion: use default port number, config field is not required, this will crate null pointer exception if user forgets to set it. As default multitasking address is set, one can assume the default port alia set as well.